include/zephyr/net/ethernet.h

```diff
@@ -347,6 +347,19 @@ enum ethernet_filter_type {
 
 /** @endcond */
 
+/** Types of Ethernet L2 */
+enum ethernet_if_types {
+	/** IEEE 802.3 Ethernet */
+	L2_ETH_IF_TYPE_ETHERNET			= BIT(0),
+
+	/** IEEE 802.11 Wi-Fi*/
+	L2_ETH_IF_TYPE_WIFI			= BIT(1),
+
+	/** IEEE 802.11 Wi-Fi Offloaded (0) or Native (1) */
+	L2_ETH_IF_TYPE_WIFI_NATIVE		= BIT(2),
```

Comment:
This is IMO ambiguous - merely the fact that the interface uses Ethernet L2 implies it's native.

Reply:
There could be a Wi-Fi chip that deals with Ethernet frames but Wi-Fi is fully offloaded, no?